File: challenge-228/steven-wilson/perl/ch-01.pl
#!/usr/bin/env perl

use 5.12.0;
use Test::More;

cmp_ok( sum_unique( ( 2, 1, 3, 2 ) ), '==', 4,  "2 unique elements" );
cmp_ok( sum_unique( ( 1, 1, 1, 1 ) ), '==', 0,  "0 unique elements" );
cmp_ok( sum_unique( ( 2, 1, 3, 4 ) ), '==', 10, "all elements unique" );
done_testing();

sub sum_unique {
    my @elements = @_;
    my %element_count;
    my $sum = 0;
    for (@elements) {
        $element_count{$_}++;
    }
    for ( keys %element_count ) {
        if ( $element_count{$_} == 1 ) {
            $sum += $_;
        }
    }
    return $sum;
}

File: challenge-228/steven-wilson/javascript/ch-01.js
"use strict";

/* Task 1: Unique Sum
You are given an array of integers.
Write a script to find out the sum of unique elements in the given array.*/

function sumUnique ( elements ) {
    let elementCount = new Map();
    let sum = 0;
    for (let elem of elements) {
        elementCount.set(elem, (elementCount.get(elem) ?? 0 ) + 1);
    }
    for (let elem of elementCount.keys()){
        if( elementCount.get(elem) == 1){
            sum += elem;
        }
    }
    return sum
}
